### PR TITLE
remove extract function in the Validation::comparison without breaking compatibility.

### PR DESCRIPTION
**Pull request by ichikaway**
This PR doesn't break compatibility. 
I've sent the pull request about same issue.
https://github.com/cakephp/cakephp/pull/7752

### CODE REVIEW

File: lib/Cake/Utility/Validation.php [hunk 1]
```diff
@@ -231,10 +231,13 @@ public static function cc($check, $type = 'fast', $deep = false, $regex = null)
  * @return bool Success
  */
 	public static function comparison($check1, $operator = null, $check2 = null) {
-		if (is_array($check1)) {
-			extract($check1, EXTR_OVERWRITE);
+		if (is_array($check1) && !isset($operator) && !isset($check2)) {
+			if (isset($check1['check1']) && isset($check1['operator']) && isset($check1['check2'])) {
```
Comment (ADmad) [inline lib/Cake/Utility/Validation.php]: Why the nested `if` instead of having all conditions in single `if`?

Comment (ichikaway) [inline lib/Cake/Utility/Validation.php]: I think it's easy to understand the aim.
Is it better to change single `if`?

Comment (ADmad) [inline lib/Cake/Utility/Validation.php]: yes

**Comment by chinpei215** [comment]
@ichikaway I prefer #7752. We need not to keep B/C about this issue. @markstory also says, "... are worth the pain".

**Comment by ichikaway** [comment]
I proposed 2 pull requests. I hope either will be merged.
IMO, It's better to keep B/C if possible.

File: lib/Cake/Utility/Validation.php (hunk 1 above)
Comment (ichikaway) [inline lib/Cake/Utility/Validation.php]: All right.
I did it.

File: lib/Cake/Utility/Validation.php [hunk 2]
```diff
@@ -231,10 +231,11 @@ public static function cc($check, $type = 'fast', $deep = false, $regex = null)
  * @return bool Success
  */
 	public static function comparison($check1, $operator = null, $check2 = null) {
-		if (is_array($check1)) {
-			extract($check1, EXTR_OVERWRITE);
+		if (is_null($operator) && is_null($check2) && isset($check1['check1']) && isset($check1['operator']) && isset($check1['check2'])) {
```
Comment (dereuromark) [inline lib/Cake/Utility/Validation.php]: Is_null() we dont use. Use strict === null instead.

Comment (ichikaway) [inline lib/Cake/Utility/Validation.php]: All right.

**Comment by lorenzo** [comment]
I personally think we should get rid of the array argument feature. It was proven to be insecure

**Pull request closed without merging**